app/javascript/helpers/Axios.jsx

```diff
@@ -30,4 +61,11 @@ const axiosInstance = axios.create(
   },
 );
 
+// Intercepts requests, filters out and forward search params to the API.
+axiosInstance.interceptors.request.use((_config) => {
+  const params = new URLSearchParams(window.location.search);
+  mergeConfigParams(filterParams(params), _config.params);
+  return { ..._config, ...{ params } };
+}, (error) => Promise.reject(error));
+
```

Comment:
I think these changes need to be extracted into their own PRs with an explanation as to what and why these changes are needed. It's hard to follow a PR that makes sudden changes like this without any information

Reply:
Details are given in #3586